src/input/mpegts/iptv/iptv_rtsp.c

```diff
@@ -89,26 +96,92 @@ iptv_rtsp_header ( http_client_t *hc )
     return 0;
   }
 
-  if (hc->hc_code != HTTP_STATUS_OK) {
+  if (hc->hc_cmd != RTSP_CMD_DESCRIBE && hc->hc_code != HTTP_STATUS_OK) {
     tvherror(LS_IPTV, "invalid error code %d for '%s'", hc->hc_code, im->mm_iptv_url_raw);
     return 0;
   }
 
+  if (hc->hc_cmd == RTSP_CMD_DESCRIBE && hc->hc_code != HTTP_STATUS_OK && hc->hc_code != HTTP_STATUS_SEE_OTHER) {
+    tvherror(LS_IPTV, "DESCRIBE request returned an invalid error code (%d) for '%s', "
+             "fall back to GET_PARAMETER in keep alive loop.", hc->hc_code, im->mm_iptv_url_raw);
+    hc->hc_rtsp_keep_alive_cmd = RTSP_CMD_GET_PARAMETER;
+    return 0;
+  }
+
   rp = im->im_data;
+  if(rp == NULL)
+    return 0;
 
   switch (hc->hc_cmd) {
+  case RTSP_CMD_DESCRIBE:
+    if(rp->play) {
+      // Already active, most probably a keep-alive response
+      break;
+    }
+    if (hc->hc_code == HTTP_STATUS_SEE_OTHER) {
+      if (!hc->hc_handle_location) {
+        tvherror(LS_IPTV, "received code 303 from RTSP server but redirects disabled '%s'",
+            im->mm_iptv_url_raw);
+        return -1;
+      }
+      // Redirect from RTSP server, parse new location and use that instead
+      p = http_arg_get(&hc->hc_args, "Location");
+      if (p == NULL) {
+        tvherror(LS_IPTV, "received code 303 from RTSP server but no new location given for '%s'",
+            im->mm_iptv_url_raw);
+        return -1;
+      }
+      tvhinfo(LS_IPTV, "received new location from RTSP server '%s' was '%s'", p,
+          im->mm_iptv_url_raw);
+      urlinit(&url);
+      if (urlparse(p ? : "", &url) || strncmp(url.scheme, "rtsp", 4) != 0) {
```

Comment:
p can not be NULL here as in line 132 it would have returned for `p == NULL`. So the `p ? : ""` can be replaced by a simple `p`.